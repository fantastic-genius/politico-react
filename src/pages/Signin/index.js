import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import NavBar from '@components/NavBar';
import Button from '@components/Button';
import InputForm from '@components/InputForm';
import { signin as signinAction } from '@actions/Auth'
import { callToast } from '@components/Toast'

const Signin = ({
  fetching, error, fetched, history, ...props
}) => {
  const initialValues = {
    email: '',
    password: '',
  }

  const [values, setValues] = useState(initialValues);

  const onChange = (e) => {
    e.persist();
    setValues(prevState => ({
      ...prevState,
      [e.target.name]: e.target.value
    }));
  }

  const onSignin = (e) => {
    e.preventDefault();
    props.signinAction(values);
  }

  useEffect(() => {
    if(error){
      callToast('error', error);
    }

    if(fetched){
      callToast('success', 'Login Successful');
      history && history.push('/')
    }

  }, [error, fetched])

  return (
    <div className="container home-con">
      <NavBar />
      <div className="item-box">
        <div className="header">
          <h2>Sign In</h2>
        </div>
        <form id="signup-form" onSubmit={ onSignin }>
          <InputForm 
            id='email'
            type='email'
            labelName='Email'
            placeholder='Email'
            name='email'
            pattern="^[\w.]+@[\w]{2,20}.[a-z]{2,10}$"
            title="must be a valid email"
            onChange={ onChange }
            required
          />
          <InputForm 
            id='password'
            type='password'
            labelName='Password'
            name='password'
            pattern="^[\w]{8,20}$"
            title="password is required and must be more than 8 character"
            autoComplete="off"
            onChange={ onChange }
            required
          />
          <div className="btn-con">
            <Button type='submit' id='login' className='btn btn-success' value='Login' />
          </div>
        </form>
      </div>
    </div>
  )
};

const mapStateToProps = (state) => ({
  fetching: state.auth.fetching,
  error: state.auth.error,
  fetched: state.auth.fetched,
  user: state.auth.user
});


export default connect(mapStateToProps, {signinAction})(Signin);
