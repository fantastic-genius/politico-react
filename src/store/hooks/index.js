import { useEffect } from 'react';
import  { setUser } from '@actions/Auth';

const useSetUser = ({ dispatch }) => {
  useEffect(() => {
    dispatch(setUser());
  }, []);
};

export default useSetUser;