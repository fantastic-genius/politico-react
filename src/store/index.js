import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { createLogger as logger } from 'redux-logger';
import promise from 'redux-promise-middleware';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import reducer from './reducers';

const middleware = applyMiddleware(promise, thunk, logger());
const store = createStore(reducer, composeWithDevTools(middleware));

export default store;
