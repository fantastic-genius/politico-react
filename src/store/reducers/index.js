import { combineReducers } from 'redux';
import auth from '@reducers/Auth';
import vote from '@reducers/Vote';
import office from '@reducers/Office';

const reducers = combineReducers({
    auth,
    vote,
    office,
});

export default reducers;
