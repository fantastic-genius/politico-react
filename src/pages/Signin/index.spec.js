import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom'
import store from '@src/store';
import Signin from './index';

describe('<Signin />', () => {
  describe('Rendering', () => {
    it('should render signin page correctly', () => {
      const component = shallow(
        <Router>
          <Signin/>
        </Router>
      );
      expect(component).toMatchSnapshot();
    });

    it('should render signin page correctly woth its children', () => {
        const component = mount(
          <Router>
            <Signin store={ store }/>
          </Router>
        );
        expect(component).toMatchSnapshot();
        expect(component.find('button')).toHaveLength(3);
        expect(component.find('input')).toHaveLength(2);
    });
  });

})
