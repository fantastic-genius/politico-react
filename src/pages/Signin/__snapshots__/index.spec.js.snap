// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signin /> Rendering should render signin page correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Connect(Signin) />
</Router>
`;

exports[`<Signin /> Rendering should render signin page correctly with its children 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Signin)
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <Signin
          error={null}
          fetched={false}
          fetching={false}
          signinAction={[Function]}
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
          user={Object {}}
        >
          <div
            className="container home-con"
          >
            <Connect(NavBar)>
              <NavBar
                isAuthenticated={false}
              >
                <header>
                  <div
                    className="app-name"
                  >
                    <Link
                      to="/"
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        <h2>
                          POLITICO
                        </h2>
                      </a>
                    </Link>
                  </div>
                  <div
                    className="sign-btn"
                  >
                    <span
                      className="home-btn"
                    >
                      <span
                        className="home-btn"
                      >
                        <Link
                          to="/signup"
                        >
                          <a
                            href="/signup"
                            onClick={[Function]}
                          >
                            <Button
                              className="btn nav-btn"
                              value="Sign Up"
                            >
                              <button
                                className="btn nav-btn"
                              >
                                Sign Up
                              </button>
                            </Button>
                          </a>
                        </Link>
                        <Link
                          to="/login"
                        >
                          <a
                            href="/login"
                            onClick={[Function]}
                          >
                            <Button
                              className="btn nav-btn"
                              value="Login"
                            >
                              <button
                                className="btn nav-btn"
                              >
                                Login
                              </button>
                            </Button>
                          </a>
                        </Link>
                      </span>
                    </span>
                  </div>
                </header>
              </NavBar>
            </Connect(NavBar)>
            <div
              className="item-box"
            >
              <div
                className="header authform-header"
              >
                <h2>
                  Sign In
                </h2>
              </div>
              <form
                id="signup-form"
                onSubmit={[Function]}
              >
                <InputForm
                  id="email"
                  labelName="Email"
                  name="email"
                  onChange={[Function]}
                  pattern="^[\\\\w.]+@[\\\\w]{2,20}.[a-z]{2,10}$"
                  placeholder="Email"
                  required={true}
                  title="must be a valid email"
                  type="email"
                >
                  <div>
                    <label
                      htmlFor="email"
                    >
                      Email
                    </label>
                    <input
                      id="email"
                      name="email"
                      onChange={[Function]}
                      pattern="^[\\\\w.]+@[\\\\w]{2,20}.[a-z]{2,10}$"
                      placeholder="Email"
                      required={true}
                      title="must be a valid email"
                      type="email"
                    />
                  </div>
                </InputForm>
                <InputForm
                  autoComplete="off"
                  id="password"
                  labelName="Password"
                  name="password"
                  onChange={[Function]}
                  pattern="^[\\\\w]{8,20}$"
                  required={true}
                  title="password is required and must be more than 8 character"
                  type="password"
                >
                  <div>
                    <label
                      htmlFor="password"
                    >
                      Password
                    </label>
                    <input
                      autoComplete="off"
                      id="password"
                      name="password"
                      onChange={[Function]}
                      pattern="^[\\\\w]{8,20}$"
                      required={true}
                      title="password is required and must be more than 8 character"
                      type="password"
                    />
                  </div>
                </InputForm>
                <div
                  className="btn-con"
                >
                  <Button
                    className="btn btn-success"
                    id="login"
                    type="submit"
                    value="Login"
                  >
                    <button
                      className="btn btn-success"
                      id="login"
                      type="submit"
                    >
                      Login
                    </button>
                  </Button>
                </div>
              </form>
            </div>
          </div>
        </Signin>
      </Connect(Signin)>
    </Router>
  </BrowserRouter>
</Provider>
`;
